Yoast/wordpress-seo
# Load xsl stylesheet inline

## Summary

This PR can be summarized in the following changelog entry:

* Makes the XSL stylesheet static, which means the jQuery tablesorting and reference to the sitemap index are removed (because these are dynamically applied).

## Test instructions

This PR can be tested by following these steps:

* First of all: remove all transients, to make sure no cache has been loaded.
* Checkout trunk
* Open the sitemap_index.xml and see tablesorting is working
* Go to a post sitemap and see a link to the Sitemap index
* Remove all transients again
* Checkout this branch and see: the table sorting not working.
* Also verify the Sitemap index link is removed on the post sitemap.

### Home/Site URL mismatch

Whenever the home URL and site URL are set to different values (local.wordpress.test/local.wordpress-home.test) the XSL needs to be loaded via the previously implemented method.
Otherwise the domain, protocol or port will mismatch and the XSL cannot be loaded.
* Configure home URL to be different from site URL and see the XSL being loaded from the home URL base (`{home_url}/main-sitemap.xsl`)

### Addons
* Also verify the sitemaps for local, news and video will still work as before. (Separate pulls will follow for these plugins)

## Quality ensurance

* [x] N/A. I have added unittests to verify the code works as intended

Fixes #8521
